src/App.js

```diff
@@ -8,46 +8,101 @@ import AddTodoForm from './addTodoForm';
 function App() {
 
   const [todoList, setTodoList] = useState([])
-  const initialList = JSON.parse(localStorage.getItem('savedTodoList')) || []
-  const [isloading,setIsLoading] = useState(true)
-
-  useEffect(() =>{
-     new Promise((resolve,reject)=>{
-      setTimeout(() => {
-        resolve({ data: { todoList: initialList } });
-      }, 2000);
-     })
-     .then(result =>{
-       setTodoList(result.data.todoList)
-       setIsLoading(false)
-     })
-  }, [])
+  // let initialList = JSON.parse(localStorage.getItem('savedTodoList')) ?? []
+  const [isloading, setIsLoading] = useState(true)
 
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: { 'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(url, options)
 
-  useEffect(() => {
-    if (!isloading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.Title, id: todo.id, createdAt: todo.createdTime }
+      })
+      const newTodos = todos.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
+      setTodoList(newTodos)
+      setIsLoading(false)
+
+    } catch (error) {
+      console.log(error.message)
     }
-  }, [todoList])
+  }
+
+  const addTodo = async (url, data) => {
```

Comment:
What was your reasoning for adding a `url` parameter? In the other callbacks defined above and below, the `url` was defined above the try/catch block, but for this function, it's defined within an argument.

Reply:
As I was writing the addTodo function it seemed to be a lot of code and the since the instructions said to pass in a url as an argument to the fetch then I decided to add the url from the addTodoForm since it was in a different file.